docs/source/wrappers.rst

```diff
@@ -0,0 +1,499 @@
+
+Only when the :py:func:`~scikeras.wrappers.KerasClassifier.fit` or
+:py:func:`~scikeras.wrappers.KerasRegressor.fit` method are called, are the
+different attributes of the wrapper, such as the ``model``, initialized.
+An initialized attribute's name always ends on an underscore; e.g., the
+initialized ``model`` is called ``model_``. (This is the same
+nomenclature as sklearn uses.) Therefore, you always know which
+attributes you set and which ones were created by the wrappers.
+
+Once initialized by calling ``fit``, the wrappers create several attributes,
+documented in the :ref:`scikeras-api` documentation.
+
+Compilation of ``Model``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+You have two options to compile your model:
+
+1. Compile your model within ``model_build_fn`` and return this
+compiled model. In this case, SciKeras will not re-compile your model
+and all compilation parameters (such as ``optimizer``) given to
+:py:func:`scikeras.wrappers.BaseWrapper.__init__` will be ignored.
+
+2. Return an uncompiled model from ``model_build_fn`` and let
+SciKeras handle the compilation. In this case, SciKeras will
+apply all of the compilation parameters, including instantiating
+losses, metrics and optimizers.
+
+The first route will be more flexible if you wish to determine how to compile
+your ``Model`` within the same function in which you define it. The latter will
+offer an easy way to compile and tune compilation parameters. Examples:
+
+.. code:: python
+
+    def model_build_fn(compile_kwargs):
+        # you can access the ``optimizer`` param here
+        optimizer = compile_kwargs["optimizer"]
+        if optimizer is None:
+            # and apply any custom logic you wish
+            ...
+        model = Model(...)
+        ...
+        model.compile(optimizer=optimizer)
+        return model
+
+    clf = KerasClassifier(model=model_build_fn)
+    clf.fit(X, y)
+    y_pred = clf.predict(X_valid)
+
+.. code:: python
+
+    from tensorflow.keras.optimizers import Adam
+
+    def model_build_fn():
+        model = Model(...)
+        ...
+        # Do not call model.compile
+        return model  # That's it, SciKeras will compile your model for you
+
+    clf = KerasClassifier(model=model_build_fn, optimizer=Adam)
+    clf.fit(X, y)
+    y_pred = clf.predict(X_valid)
+
+
+In all cases, returning an un-compiled model is equivalent to
+calling ``model.compile(**compile_kwargs)`` within ``model_build_fn``.
+
+
+Arguments to ``model_build_fn``
+-------------------------------
+
+User-defined keyword arguments passed to :py:func:`~scikeras.wrappers.BaseWrapper.__init__`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+All keyword arguments that were given to :py:func:`~scikeras.wrappers.BaseWrapper.__init__`
+will be passed to ``model_build_fn`` directly.
+For example, calling ``KerasClassifier(myparam=10)`` will result in a
+``model_build_fn(my_param=10)`` call.
```

Comment:
I thought this behavior was deprecated. Shouldn't it be `model__myparam`?

Reply:
As per https://github.com/adriangb/scikeras/pull/67#issuecomment-685114298 we need to keep this for backward compatibility with the old API. So I think the _only_ benefit of having the `model__` prefix is that:
(1) At some point we can make it required, after the `1.0` release or something
(2) It can be used to for example have `warm_start` and `model__warm_start` or something (I don't see why that would be necessary, but it is possible).